packages/vite-plugin/src/index.ts

```diff
@@ -100,7 +108,11 @@ export function vanillaExtractPlugin({
         return;
       }
 
-      if (!server || server.config.isProduction) {
+      if (
+        !server ||
+        server.config.isProduction ||
+        validId.endsWith(virtualExtCss)
```

Comment:
I think the first two checks are pointless now. If we're loading a `.vanilla.css` file then it should always be a straight CSS return.